src/common/class.wpes-wp-post-field-builder.php

```diff
@@ -534,6 +534,127 @@ public function get_mappings( $args = array() ) {
+						'mime' => array(
+							'type' => 'string',
+							'index' => 'not_analyzed',
+						),
+						'mime_type' => array(
+							'type' => 'string',
+							'index' => 'not_analyzed',
+						),
+						'mime_subtype' => array(
+							'type' => 'string',
+							'index' => 'not_analyzed',
+						),
+						// This is post_title in the DB
+						'title'	=> array(
+							'type' => 'multi_field',
+							'fields' => array(
+								'title' => array(
+									'type' => 'string',
+									'index' => 'analyzed',
+									'similarity' => 'BM25',
+								),
+								'word_count' => array(
+									'type' => 'token_count',
+									'analyzer' => 'default',
+								),
+							),
+						),
+						// This is post_content in the DB
+						'description' => array(
+							'type' => 'multi_field',
+							'fields' => array(
+								'content' => array(
+									'type' => 'string',
+									'index' => 'analyzed',
+									'similarity' => 'BM25',
+								),
+								'word_count' => array(
+									'type' => 'token_count',
+									'analyzer' => 'default',
+								),
+							),
+						),
+						// This is post_excerpt in the DB
+						'caption' => array(
+							'type' => 'multi_field',
+							'fields' => array(
+								'excerpt' => array(
+									'type' => 'string',
+									'index' => 'analyzed',
+									'similarity' => 'BM25',
+								),
+								'word_count' => array(
+									'type' => 'token_count',
+									'analyzer' => 'default',
+								),
+							),
+						),
+						// @question Did I construct this correctly?
+						'dimensions_token' => array(
+							'type' => 'object',
+							'properties' => array(
+								'width' => array(
+									'type' => 'short',
+								),
+								'height' => array(
+									'type' => 'short',
+								),
+								'area' => array(
+									'type' => 'integer',
+								),
+							),
+						),
+						'grayscale' => array(
+							'type' => 'boolean',
+						),
+						'alpha' => array(
+							'type' => 'boolean',
+						),
+						// @note This simply stores the name of dominant color as a string
+						// Uses 16 HTML colors: http://www.wikiwand.com/en/Web_colors#/HTML_color_names
+						// + the orange color from CSS 2.1: http://www.wikiwand.com/en/Web_colors#/CSS_colors
+						//
+						// @question Do we need this? Can we query the color_token below to get the same information?
+						'primary_color_name' => array(
+							'type' => 'string',
+							'index' => 'not_analyzed',
+						),
+						// @note This stores all found color accents from N samples. Allows us to make more
+						// specfic searches, e.g. find black images with just a hint of orange and red.
+						//
+						// The script will first convert the image to a mosaic of N areas. Then it will take
+						// a sample pixel from each area and convert it to one of the 17 basic colors.
+						// The dominance field is just a percentage of samples that fell into a particular bucket.
+						'color_token' => array(
```

Comment:
@gibrown I was thinking about using [`imagefilter`](http://php.net/manual/en/function.imagefilter.php) method with the `IMG_FILTER_PIXELATE` flag to create a mosaic picture where every area has a uniform color. This could also be accomplished by shrinking the image and scaling it to the original size using `imagecopyresized`. Then it doesn't matter which pixel we sample, because they all have the same color within the area which is based on the color of all different pixels originally in that area.

When we get this color, we can simply use `Jetpack_Color->getClosestMatch()` to file it in one of our predefined buckets - those 17 colors mentioned in the code. We can prepare any buckets we like here. Then we just count how many samples ended in each bucket and convert this value into percentage representation.

I was thinking about storing a HSV value of the most dominant colors in the picture. HSV [is simply queryable](http://dpb587.me/blog/2014/04/24/color-searching-with-elasticsearch.html), one can much more easily search for a range of similar colors in this color space that it would be possible in RGB, for example. This would be possible, but it also is quite tricky as the problem of color quantization, while in theory very straightforward, clashes with specifics of human vision.